tests/writer_test.py

```diff
@@ -0,0 +1,58 @@
+# -*- coding: utf-8 -*-
+"""Tests for the artifact definitions readers."""
+
+import io
+import os
+import unittest
+import json
+import tempfile
+
+from artifacts import reader
+from artifacts import writer
+
+
+class YamlArtifactsWriterTest(unittest.TestCase):
+  """Class to test the YAML artifacts writer"""
+
+  def testYamlWriter(self):
+    """Tests the YamlArtifactsWriter FormatArtifacts method for loss during conversion."""
+    artifact_reader = reader.YamlArtifactsReader()
+    artifact_writer = writer.YamlArtifactsWriter()
+    test_file = os.path.join('test_data', 'definitions.yaml')
+
+    artifact_definitions = list(artifact_reader.ReadFile(test_file))
+    with tempfile.NamedTemporaryFile() as artifact_file:
+      artifact_writer.WriteArtifactsFile(artifact_definitions,
+                                         artifact_file.name)
+      converted_artifact_definitions = list(artifact_reader.ReadFile(
+          artifact_file.name))
+
+    self.assertListEqual(
+        [artifact.AsDict() for artifact in artifact_definitions],
+        [artifact.AsDict() for artifact in converted_artifact_definitions])
+
+
+class JsonArtifactsWriterTest(unittest.TestCase):
+  """Class to test the JSON artifacts writer"""
+
+  def testJsonWriter(self):
+    """Tests the JsonArtifactsWriter FormatArtifacts method for loss during conversion."""
+    artifact_reader = reader.YamlArtifactsReader()
+    artifact_writer = writer.JsonArtifactsWriter()
+    test_file = os.path.join('test_data', 'definitions.yaml')
+
+    artifact_definitions = list(artifact_reader.ReadFile(test_file))
```

Comment:
The reason for this is a lack of JsonArtifactReader.  I can write a json file to disk, but then the only code in the project that actually reads an artifact definition from a json file would be in the test so I just skipped the writing piece.

Because the yaml.safe_load_all() operates differently than json.loads() reading with 'rb' for json.loads() causes more py2/3 issues in writing a new JsonArtifactReader for a format that isn't supposed to really be on disk as JSON anyways.